### ISSUE TITLE
asm labelling docs unclear due to ignored var

### ISSUE THREAD

**Issue by zeemy23** [terraform-google-modules/terraform-google-kubernetes-engine#1671]
### TL;DR

https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/master/modules/asm/README.md

These docs guide the user to add the mesh_id label in a way that this PR from 2021 discontinued support for. 

 https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1061

I can add a PR that clarifies the readme for future users. We were pulling our hair out trying to figure out why the label was getting ignored.

### Expected behavior

following asm docs results in asm functionality

### Observed behavior

ignored labels causes much strife if following docs

### Terraform Configuration

```hcl
module "gke" {
...
  cluster_resource_labels = { "mesh_id" : "proj-${data.google_project.project.number}" }
...
}
```

### Terraform Version

```sh
all versions.
```

**Comment by legal90** [comment]
That issue is still actual. Should it be reopened? cc @apeabody , @ferrarimarco 

Both ASM tf module documentation, as wells as the official Google docs are saying that `mesh_id` label has to be added explicitly to the cluster:
- https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/3c9a634ceef3fe8fc073de161f17f0064f424c04/modules/asm/README.md
- https://cloud.google.com/service-mesh/docs/managed/provision-managed-anthos-service-mesh#apply_the_mesh_id_label

But, as @zeemy23 said above, this label is ignored on gke tf modules:
https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/eeebd782df42bc197638f4fd291ee99a680002fc/autogen/main/cluster.tf.tmpl#L248-L250
It was added by https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1061.

**Comment by apeabody** [comment]
> That issue is still actual. Should it be reopened? cc @apeabody , @ferrarimarco
> 
> Both ASM tf module documentation, as wells as the official Google docs are saying that `mesh_id` label has to be added explicitly to the cluster:
> 
> * https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/3c9a634ceef3fe8fc073de161f17f0064f424c04/modules/asm/README.md
> * https://cloud.google.com/service-mesh/docs/managed/provision-managed-anthos-service-mesh#apply_the_mesh_id_label
> 
> But, as @zeemy23 said above, this label is ignored on gke tf modules:
> 
> https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/eeebd782df42bc197638f4fd291ee99a680002fc/autogen/main/cluster.tf.tmpl#L248-L250
> 
> 
> It was added by #1061.

Removed the stale tag and reopened.  cc @ferrarimarco who was working on the ASM sub-module.

**Comment by ferrarimarco** [comment]
Hi! Feel free to propose a PR. I'll review it :)

Thanks!

**Comment by legal90** [comment]
@ferrarimarco Thank you! Here is the PR: https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1836
